pyqtgraph/Point.py

```diff
@@ -87,47 +93,67 @@ def __pow__(self, a):
         return self._math_('__pow__', a)
     
     def _math_(self, op, x):
-        x = Point(x)
-        return Point(getattr(self[0], op)(x[0]), getattr(self[1], op)(x[1]))
+        if not isinstance(x, QtCore.QPointF):
+            x = Point(x)
+        return Point(getattr(self.x(), op)(x.x()), getattr(self.y(), op)(x.y()))
     
     def length(self):
         """Returns the vector length of this Point."""
-        return hypot(self[0], self[1])  # length
+        return hypot(self.x(), self.y())  # length
 
     def norm(self):
         """Returns a vector in the same direction with unit length."""
         return self / self.length()
     
-    def angle(self, a):
-        """Returns the angle in degrees between this vector and the vector a."""
-        rads = atan2(self.y(), self.x()) - atan2(a.y(), a.x())
+    def angle(self, a, units="degrees"):
+        """
+        Returns the angle in degrees between this vector and the vector a.
+        
+        Parameters
+        ----------
+        a : Point, QPointF or QPoint
+            The Point to return the angle with
+        units : str, optional
+            The units with which to compute the angle with, "degrees" or "radians",
+            default "degrees"
+        
+        Returns
+        -------
+        float
+            The angle between the two points
+        """
+        rads = atan2(a.y(), a.x()) - atan2(self.y(), self.x())
+        if units == "radians":
+            return rads
         return degrees(rads)
     
     def dot(self, a):
         """Returns the dot product of a and this Point."""
-        a = Point(a)
-        return self[0]*a[0] + self[1]*a[1]
+        if not isinstance(a, QtCore.QPointF):
+            a = Point(a)
+        return Point.dotProduct(self, a)
     
     def cross(self, a):
-        a = Point(a)
-        return self[0]*a[1] - self[1]*a[0]
+        if not isinstance(a, QtCore.QPointF):
+            a = Point(a)
+        return self.x() * a.y() - self.y() * a.x()
         
     def proj(self, b):
         """Return the projection of this vector onto the vector b"""
         b1 = b / b.length()
-        return self.dot(b1) * b1
+        return self.dot(b) * b1
```

Comment:
This changes the meaning. Try for instance ```pg.Point(3, 4).proj(pg.Point(0, 100))```

Reply:
that, was an unintentional change.